src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String report = calculateReport(readFromFile(fromFileName));
+            writeToFile(toFileName, report);
+            System.out.println("Report generated successfully.");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
you should better handle corresponding exceptions in `readFromFile` and `writeToFile` methods

this ways you'll be able to throw dedicated exceptions from there - you should not just print stack trace